lib/extensions.js

```diff
@@ -40,6 +40,35 @@ module.exports = function(realmConstructor) {
     setConstructorOnPrototype(realmConstructor.Results);
     setConstructorOnPrototype(realmConstructor.Object);
 
+    //Add async open API
+    Object.defineProperties(realmConstructor, getOwnPropertyDescriptors({
+        open(config) {
+            return new Promise((resolve, reject) => {
+                realmConstructor._waitForDownload(config, (error) => {
+                    if (error) {
+                        reject(error);
+                    }
+
+                    let syncedRealm = new realmConstructor(config);
+                    //FIXME: RN hangs here. Remove when node's makeCallback alternative is implemented
```

Comment:
The PR is usint the makeCallback implementation which is done as a commit in this PR. It will be merged when this PR merges.

The issue is with ReactNative where there is no makeCallback available and this hangs. The code that we use seems all fine but still we hang on this piece of code. There are still ways to implement this correctly but they need more research. Hence the FIXME comment.